File: scripts/mcp_refresh_peer.py
# SPDX-License-Identifier: Apache-2.0
import json
import os
import sys
import threading
from http.server import ThreadingHTTPServer
from psp_cdl_mcp_server import McpServer
from psp_cdl_mcp_server.stdio import serve_stdio
from psp_cdl_mcp_server.http import McpHttpServer,make_http_handler
from mcp_refresh_fixtures import TestPeer,PRINCIPAL
from http_fixtures import config
c=json.loads(sys.argv[1]);peer=TestPeer(c.get("mode","normal"));call=peer.call_tool
def counted(*args):
    with open(c["spy"],"a",encoding="utf-8") as stream:stream.write("call\n")
    return call(*args)
def authenticate(token):
    if token!="synthetic-refresh-token":raise ValueError("DENIED")
    return PRINCIPAL
peer.call_tool=counted;peer.authenticate=authenticate
if c["transport"]=="stdio":serve_stdio(McpServer(peer,lambda:os.environ.get("PSP_REFRESH_TOKEN","")))
else:
    server=ThreadingHTTPServer(("127.0.0.1",0),make_http_handler(None))
    endpoint=f"http://127.0.0.1:{server.server_port}/mcp"
    class Host:
        def authenticate(self,t,r):return PRINCIPAL if t=="synthetic-refresh-token" and r==endpoint else None
        def open(self,*_):return {"service":peer,"close":lambda:None}
    adapter=McpHttpServer(Host(),config(endpoint));server.RequestHandlerClass=make_http_handler(adapter)
    thread=threading.Thread(target=server.serve_forever,daemon=True);thread.start()
    print(endpoint,flush=True)
    sys.stdin.read();adapter.close();server.shutdown();server.server_close();thread.join()

File: scripts/mcp-refresh-peer.mjs
// SPDX-License-Identifier: Apache-2.0
// Synthetic authenticated peer for actual transport interchange tests.
import {appendFileSync} from 'node:fs';
import {createServer} from 'node:http';
import {McpServer} from '@psp-cdl/mcp-server';
import {serveStdio} from '@psp-cdl/mcp-server/stdio';
import {McpHttpServer,nodeHttpHandler} from '@psp-cdl/mcp-server/http';
import {TestPeer,principal} from './mcp-refresh-fixtures.mjs';
import {config} from './http-fixtures.mjs';
const c=JSON.parse(process.argv[2]),peer=new TestPeer(c.mode),call=peer.service.callTool;
peer.service.callTool=async(...args)=>{appendFileSync(c.spy,'call\n');return call(...args);};
peer.service.authenticate=t=>{if(t!=='synthetic-refresh-token')throw Error('DENIED');return principal;};
if(c.transport==='stdio')await serveStdio(new McpServer(peer.service,()=>process.env.PSP_REFRESH_TOKEN??''));
else{
  let adapter;
  const server=createServer((req,res)=>nodeHttpHandler(adapter)(req,res));
  await new Promise(r=>server.listen(0,'127.0.0.1',r));
  const endpoint=`http://127.0.0.1:${server.address().port}/mcp`;
  adapter=new McpHttpServer({authenticate:(t,r)=>t==='synthetic-refresh-token'&&r===endpoint?principal:null,open:()=>({service:peer.service,close:()=>{}})},config(endpoint));
  console.log(endpoint);process.stdin.resume();process.stdin.on('end',()=>{adapter.close();server.closeAllConnections();server.close();});
}
